### PR TITLE
ChangeBlobber unit test

### LINKED ISSUE #1216: Unit test for change blobber
The unit tests we need to cover including:
- Blobber not found, assert error
- Blobber has no allocation, assert error
- Blobber has one allocation, and then removed, assert related partitions and MPT nodes are all removed or updated appropriately.
- Blobber has more than one allocation, assert that the blobber allocation partition is update, all others remain untouched.

### PR DESCRIPTION
**Pull request by ssardana08**
## Changes
- Unit test for changeBlobber
- Unit test for removeBlobber
- Fix removeBlobber logical error

## Tests 
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/0chain/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

## Associated PRs (Link as appropriate):
- blobber:
- gosdk:
- system_test:
- zboxcli:
- zwalletcli:
- Other: ...

### CODE REVIEW

File: code/go/0chain.net/smartcontract/storagesc/allocation_test.go [hunk 1]
```diff
@@ -324,7 +539,7 @@ func TestExtendAllocation(t *testing.T) {
 			).Return(nil).Once()
 		}
 
-		balances.On("EmitEvent", event.TypeStats, event.TagUpdateBlobber, mock.Anything, mock.Anything).Return()
+		//balances.On("EmitEvent", event.TypeStats, event.TagUpdateBlobber, mock.Anything, mock.Anything).Return()
```
Comment (Sriep) [inline code/go/0chain.net/smartcontract/storagesc/allocation_test.go]: Remove comment

File: code/go/0chain.net/smartcontract/storagesc/models.go [hunk 2]
```diff
@@ -886,15 +886,15 @@ func (sa *StorageAllocation) removeBlobber(
...
-	}
+	//blobber, err := ssc.getBlobber(removeId, balances)
+	//if err != nil {
+	//	return nil, err
+	//}
+	//blobber.Used -= sa.bSize()
+	//_, err = balances.InsertTrieNode(blobber.GetKey(ssc.ID), blobber)
+	//if err != nil {
+	//	return nil, err
+	//}
```
Comment (Sriep) [inline code/go/0chain.net/smartcontract/storagesc/models.go]: Remove commented-out code.

File: code/go/0chain.net/smartcontract/storagesc/allocation_test.go [hunk 3]
```diff
@@ -230,6 +232,219 @@ func TestSelectBlobbers(t *testing.T) {
...
+			name: "test_blobber_add_remove_blobber",
+			args: args{
+				numBlobbers:     6,
+				addBlobberID:    "add_blobber_id",
+				removeBlobberID: "blobber_1",
+				dataShards:      5,
+			},
+			want: want{
+				err: false,
+			}},
```
Comment (peterlimg) [inline code/go/0chain.net/smartcontract/storagesc/allocation_test.go:517]: Please add more test cases, we should cover most of the process logic of the changeBlobbers method.

File: code/go/0chain.net/smartcontract/storagesc/allocation_test.go [hunk 4]
```diff
@@ -230,6 +232,219 @@ func TestSelectBlobbers(t *testing.T) {
...
+
+	}
+
+	testCases := []struct {
+		name string
+		args args
+		want want
+	}{
+		{
+			name: "test_blobber_remove_doesnt_exist",
```
Comment (peterlimg) [inline code/go/0chain.net/smartcontract/storagesc/allocation_test.go]: We don't need the `test_` prefix of the name

File: code/go/0chain.net/smartcontract/storagesc/allocation_test.go [hunk 5]
```diff
@@ -230,6 +232,219 @@ func TestSelectBlobbers(t *testing.T) {
...
+			want: want{
+				err: false,
+			}},
+	}
+
+	for _, tt := range testCases {
+		t.Run(tt.name, func(t *testing.T) {
+			blobbers, addID, removeID, sc, sa, now, balances := setup(tt.args)
+			_, err := sa.changeBlobbers(blobbers, addID, removeID, sc, now, balances)
+			require.EqualValues(t, tt.want.err, err != nil)
```
Comment (peterlimg) [inline code/go/0chain.net/smartcontract/storagesc/allocation_test.go:536]: Please assert the state as well, if we remove a blobber, then we need to check and assert the blobber is removed, and related state were updated correctly.

**Review by peterlimg — CHANGES_REQUESTED** [review]

**Review by Sriep — APPROVED** [review]

**Review by peterlimg — APPROVED** [review]
lgtm

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Pull request merged**